packages/xchain-ethereum/src/client.ts

```diff
@@ -424,6 +435,76 @@ export default class Client extends BaseXChainClient implements XChainClient, Et
     }
   }
 
+  /**
+   * Transaction to THORChain inbound address.
+   *
+   * @param {DepositParams} params The transaction options.
+   * @returns {TxHash} The transaction hash.
+   *
+   * @throws {"halted chain"} Thrown if chain is halted.
+   * @throws {"halted trading"} Thrown if trading is halted.
+   * @throws {"amount is not approved"} Thrown if the amount is not allowed to spend
+   * @throws {"router address is not defined"} Thrown if router address is not defined
+   */
+  async deposit({ walletIndex = 0, asset = AssetETH, amount, memo }: DepositParams): Promise<TxHash> {
+    const inboundDetails = await getInboundDetails(asset.chain, this.network)
+
+    if (inboundDetails.haltedChain) {
+      throw new Error('halted chain')
+    }
+    if (inboundDetails.haltedTrading) {
+      throw new Error('halted trading')
+    }
+    if (!inboundDetails.router) {
+      throw new Error('router address is not defined')
+    }
+
+    const abi = TCRopstenAbi
```

Comment:
Why abi from Ropsten, but not from mainnet?

Reply:
It works correct for mainnet and testnet